packages/network/src/types.js

```diff
@@ -188,3 +188,10 @@
  * ) => PromiseVow<Connection>} outbound
  *   Create an outbound connection
  */
+
+/**
+ * @typedef {object} PortAllocator
+ * @property {() => PromiseVow<Port>} allocateIBCPort
+ * @property {() => PromiseVow<Port>} allocateICAControllerPort
+ * @property {() => PromiseVow<Port>} allocateIBCPegasusPort
+ */
```

Comment:
```suggestion
/** @typedef {ReturnType<preparePortAllocator>} PortAllocator */
```

For automatic type updates if we add a new method

Reply:
To implement @0xpatrickdev's suggestion, you'll actually need to further unwrap the return type:
```suggestion
/** @typedef {ReturnType<ReturnType<typeof preparePortAllocator>>} PortAllocator */
```

Reply:
Yes, ty @michaelfig